src/main/java/net/smartcosmos/dao/metadata/converter/MetadataCreateToMetadataEntityArrayConverter.java

```diff
@@ -0,0 +1,43 @@
+package net.smartcosmos.dao.metadata.converter;
+
+import net.smartcosmos.dao.metadata.domain.MetadataEntity;
+import net.smartcosmos.dao.metadata.util.UuidUtil;
+import net.smartcosmos.dto.metadata.MetadataCreate;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.format.FormatterRegistrar;
+import org.springframework.format.FormatterRegistry;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+
+@Component
+public class MetadataCreateToMetadataEntityArrayConverter
+        implements Converter<MetadataCreate, MetadataEntity[]>, FormatterRegistrar {
+
+    @Override
+    public MetadataEntity[] convert(MetadataCreate entity) {
+
+        MetadataEntity[] metadataEntities = new MetadataEntity[entity.getMetadata().size()];
+
+        int i = 0;
+        for (Map.Entry<String, Object> entry : entity.getMetadata().entrySet()) {
+
+            MetadataEntity newEntity = MetadataEntity.builder()
+                .ownerType(entity.getOwnerType())
+                .ownerId(UuidUtil.getUuidFromUrn(entity.getOwnerUrn()))
+                .dataType(entry.getValue() != null ? entry.getValue().getClass().getSimpleName() : "null")
```

Comment:
This might give you wrong types. Remember: The types are implicit from JSON, and JSON knows only very particular types. I think, we should particularly test for those.